src/v/kafka/server/usage_manager.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "config/property.h"
+#include "storage/kvstore.h"
+
+#include <seastar/core/gate.hh>
+#include <seastar/core/timer.hh>
+
+namespace kafka {
+
+/// Main structure of statistics that are being accounted for. These are
+/// periodically serialized to disk, hence why the struct inherits from the
+/// serde::envelope
+struct usage
```

Comment:
Maybe append `_counter` to the name?  (Assuming there will be some equivalent gauge class I guess for storage usage)

Reply:
Maybe there should be a different solution because ATM one of the 3 values within this class acts like a gauge. So maybe there should be a different organization scheme to tag these values with a metric property type